app/models/framework_response.rb

```diff
@@ -13,11 +14,19 @@ class FrameworkResponse < VersionedModel
     record.errors.add(:value, :blank) if requires_value?(value, record)
   end
 
+  after_validation :set_responded_value
+
   def self.requires_value?(value, record)
     return false if value.present? || !record.framework_question.required
 
     return true if record.parent.blank?
 
     record.parent.option_selected?(record.framework_question.dependent_value)
   end
+
+private
+
+  def set_responded_value
+    self.responded = true if value_text.present? || value_json.present?
```

Comment:
we can avoid the id
```
self.responded = value_text.present? || value_json.present?
```

Reply:
So this was intentional as this flag is set to mark if a question has been seen/answered and when set to true cannot be changed. That said I realised there is missing logic around if a question is *not* required and we do get empty values, we should allow this to be set to true. I'll amend